engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/hv/EmailValidator.java

```diff
@@ -104,17 +105,26 @@ private boolean matchPart(String part, Pattern pattern, int maxLength) {
 	private String toAscii(String unicodeString) throws IllegalArgumentException {
 		String asciiString = "";
 		int start = 0;
-		int end = unicodeString.length() <= 63 ? unicodeString.length() : 63;
+		int index = getBreakingIndex( unicodeString, ASCII_MAX_LABEL_LENGTH );
+		int end = unicodeString.length() <=  index ? unicodeString.length() : index;
```

Comment:
Couldn't you make the 2 lines above a single method instead of having 2 parts and repeating it below? Something like `getIDNToASCIIPartEnd(String unicodeString, int start)`.